plugins/discovery-file/src/main/java/org/elasticsearch/discovery/file/FileBasedDiscoveryPlugin.java

```diff
@@ -104,8 +103,8 @@ public void close() throws IOException {
     public Settings additionalSettings() {
         // For 5.0, the hosts provider was "zen", but this was before the discovery.zen.hosts_provider
         // setting existed. This check looks for the legacy zen, and sets the file hosts provider if not set
-        String discoveryType = DiscoveryModule.DISCOVERY_TYPE_SETTING.get(settings);
-        if (discoveryType.equals("zen")) {
+        if (DiscoveryModule.DISCOVERY_TYPE_SETTING.exists(settings) == false ||
```

Comment:
I got confused and I see why you added this. I think it merits an extra comment indicating that at this point the discovery type setting is not defaulted to zen. Also, I think we can remove this clause completely in 6.0. This deprecation happened in 5.1.1 (#21036)